src/components/CodeOverview/index.spec.tsx

```diff
@@ -0,0 +1,452 @@
+        ...document.createElement('div'),
+        clientHeight: 400,
+      },
+    };
+    instance.setOverviewHeight(fakeRef);
+
+    expect(root.state('overviewHeight')).toEqual(fakeRef.current.clientHeight);
+  });
+
+  it('only sets the overview height for defined refs', () => {
+    const { root, instance } = renderWithInstance();
+
+    const overviewHeight = 200;
+    root.setState({ overviewHeight });
+
+    instance.setOverviewHeight(undefined);
+
+    // Make sure no new height was set.
+    expect(root.state('overviewHeight')).toEqual(overviewHeight);
+  });
+
+  it('only sets the overview height for active refs', () => {
+    const { root, instance } = renderWithInstance();
+
+    const overviewHeight = 200;
+    root.setState({ overviewHeight });
+
+    // Set an inactive ref.
+    instance.setOverviewHeight(React.createRef());
+
+    // Make sure no new height was set.
+    expect(root.state('overviewHeight')).toEqual(overviewHeight);
+  });
+
+  it('can reset the overview height', () => {
+    const { root, instance } = renderWithInstance();
+
+    // Set a height that will be reset.
+    root.setState({ overviewHeight: 200 });
+
+    instance.resetOverviewHeight();
+
+    expect(root.state('overviewHeight')).toEqual(null);
+  });
+
+  it('does not render content until the overview height is set', () => {
+    const root = render({ content: '// pretend this is code' });
+    root.setState({ overviewHeight: null });
+
+    const innerRoot = renderWithLinterProvider({ root });
+    expect(innerRoot.find(CodeLineShapes)).toHaveLength(0);
+  });
+
+  it('renders links for a code line', () => {
+    const location = createFakeLocation();
+    const root = render({
+      content: generateFileLines({ count: 3 }).join('\n'),
+      location,
+    });
+    root.setState({ overviewHeight: 400 });
+
+    const innerRoot = renderWithLinterProvider({ root });
+
+    const line = 3;
+    const link = innerRoot.find(Link).at(line - 1);
+
+    expect(link).toHaveProp('to', {
+      ...location,
+      hash: getCodeLineAnchor(line),
+    });
+
+    expect(link).toHaveProp('title', `Jump to line ${line}`);
+  });
+
+  it('sets link styles', () => {
+    const rowHeight = 10;
+    const rowTopPadding = 2;
+
+    const root = render({
+      rowHeight,
+      rowTopPadding,
+      content: generateFileLines({ count: 3 }).join('\n'),
+    });
+    root.setState({ overviewHeight: 400 });
+
+    const innerRoot = renderWithLinterProvider({ root });
+
+    const link1 = innerRoot.find(Link).at(0);
+
+    expect(link1).toHaveStyle('height', `${rowHeight}px`);
+    // The first line should not have top padding.
+    expect(link1).toHaveStyle('paddingTop', undefined);
+
+    const link2 = innerRoot.find(Link).at(1);
+
+    expect(link2).toHaveStyle('height', `${rowHeight}px`);
+    expect(link2).toHaveStyle('paddingTop', `${rowTopPadding}px`);
+
+    const link3 = innerRoot.find(Link).at(2);
+
+    expect(link3).toHaveStyle('height', `${rowHeight}px`);
```

Comment:
I'm not sure that you need to keep repeating the assertion for `height` as it's a property that doesn't depend on the particular `Link`, but it's fine by me.

Reply:
It's just making sure that the value gets passed from props to the style attribute on every element. The `paddingTop` assertion is more important.